### PR TITLE
Fix syswrite in Net::Cmd

### PR DESCRIPTION
**Pull request by xdg**
This PR addresses the concerns I raised reopening
[rt #14875](https://rt.cpan.org/Ticket/Display.html?id=14875).

It refactors all uses of syswrite to use a common routine that handles
restarts and timeouts.

### CODE REVIEW

File: lib/Net/Cmd.pm [hunk 1]
```diff
@@ -240,17 +290,13 @@ sub command {
...
     $cmd->debug_print(1, $str)
       if ($cmd->debug);
 
-    unless (defined($swlen = syswrite($cmd,$str,$len)) && $swlen == $len) {
-      $cmd->close;
-      $cmd->_set_status_closed;
-      return $cmd;
-    }
+    # though documented to return undef on failure, the legacy behavior
+    # was to turn $cmd even on failure, so this odd construct does that
```
Comment (tmetro) [inline lib/Net/Cmd.pm]: s/turn/return/

**Comment by tmetro** [comment]
If this passes whatever tests are available, it looks like a worthy refactoring.

**Comment by steve-m-hay** [comment]
Thanks for the fix. Now merged, with a minor follow-up in https://github.com/steve-m-hay/perl-libnet/commit/afc1647a2b

**Comment by rjbs** [comment]
@steve-m-hay Know what the time frame for this release is?  It'd be good to have it fixed in perl v5.24.0 — although that's a ways off, yet. ;)

**Comment by steve-m-hay** [comment]
On 2 January 2016 at 20:01, Ricardo Signes <notifications@github.com> wrote:

> @steve-m-hay <https://github.com/steve-m-hay> Know what the time frame
> for this release is? It'd be good to have it fixed in perl v5.24.0 —
> although that's a ways off, yet. ;)
>
>
I'll make a release in the next week or so. I was holding off, hoping that
I'd have time to look at
https://rt.cpan.org/Public/Bug/Display.html?id=110617 first, but since this
is quite an important fix I won't hold it up any longer.

**Pull request merged**